### ISSUE TITLE
Patching Monaco and failing. No correct glyphs, non aligned or Powerline symbols too big

### ISSUE THREAD

**Issue by luispuerto** [ryanoasis/nerd-fonts#210]
#### Subject of the issue

I've been trying all day today to patch correctly `Monaco and Menlo Nerd Complete **Mono**`, but I've failed spectacularly. I've been trying different options provided by the script but I wasn't able to a correctly display font on my terminal. 

First, I've tried to patch the original font from my fonts folder (`System/Library/Fonts/`), but I wasn't able to make it in a way to display the correct glyphs or correctly aligned. 

Then, I found that @taohex already has done the patching, thus I downloaded the fond and tried to transform to mono, the closest thing I've been able to yield is this 

![screen shot 2017-10-23 at 15 16 04](https://user-images.githubusercontent.com/22882880/31899613-b9d97ed2-b824-11e7-8185-f51058ff385e.png)

As you can see the Powerline extra symbols are slightly bigger than the fond, and I can't figure out how to fix it. 

I've got the same result with the `Menlo` font. 

If you can lend me a hand I would be grateful. I also think that, if possible, if you can add those fonts to the repository would be great. `Monaco` is the default font of rStudio and now it's possible to call the terminal from within the app so I would like to use zsh with nerd fonts to make similar to my setup in iTerm2. 

#### Your Setup

- I'm trying to patch mainly `Monaco` and also `Menlo`. 
- I use `iTerm2` as my main terminal, but I'm trying to patch `Monaco` because I want to use it on `rStudio`. I need the fonts in mono because rStudio uses only mono fonts. 
- I'm using `Mac OS X 10.11.6 El Capitan` and I've dowloaded `fontforge` with `homebrew`.

**Comment by ryanoasis** [comment]
@luisspuerto Thanks for the detailed report :+1: 

So does it look fine in the non-mono form?

You tried using [@taohex's patched Monaco](https://github.com/taohex/fonts/blob/master/Monaco%20for%20Powerline%20Nerd%20Font%20Complete.otf) but when you try to patch as mono it looks as shown in your screenshot?

**Comment by luisspuerto** [comment]
Hi!! 

If I make myself a non-mono form the result is exactly the same. 

@taohex's patched monaco look's good in iTerm
![screen shot 2017-10-27 at 08 50 57](https://user-images.githubusercontent.com/22882880/32089697-264c02d4-baf4-11e7-828b-9f3bb40ab639.png)

but mine doesn't 
![screen shot 2017-10-27 at 08 51 08](https://user-images.githubusercontent.com/22882880/32089705-3353d47a-baf4-11e7-9550-dbcb45e1d7db.png)

If you look closely... mine is like the height of the line shrinks a little bit. This is the line size of the original Monaco. 
![screen shot 2017-10-27 at 08 51 18](https://user-images.githubusercontent.com/22882880/32089731-6b3ecab6-baf4-11e7-8c77-25cd1ed54d0a.png)

It's something that can be noticed in the mac font book app. 

![screen shot 2017-10-27 at 08 54 54](https://user-images.githubusercontent.com/22882880/32089758-921be2fe-baf4-11e7-8ec6-96678e99e1a8.png)
![screen shot 2017-10-27 at 08 55 00](https://user-images.githubusercontent.com/22882880/32089759-92464f30-baf4-11e7-87c5-baae4c7aea25.png)
![screen shot 2017-10-27 at 08 55 05](https://user-images.githubusercontent.com/22882880/32089760-92710996-baf4-11e7-9431-17829a92f752.png)

Any idea of how to fix?

**Comment by luisspuerto** [comment]
I did the patching on a Raspberry Pi with Raspbian Stretch, coping the Monaco from my Mac, and de result was the same I got when I past directly on my mac. With the wrong glyphs and not correctly aligned... 

![screen shot 2017-10-27 at 09 22 18](https://user-images.githubusercontent.com/22882880/32090578-bfa211b8-baf8-11e7-984d-f6f1eb413355.png)

Besides, I tried to do a second time and give me a lot of error. You can see the log files here: 

https://gist.github.com/luisspuerto/8e8e31a8ae6b92b3ba5f855ee2249b99

The output for the same time in the pi was the same as in the mac. In the mac never ever return me an error.

**Comment by luisspuerto** [comment]
PS/ when I saw the un aligned glyphs and the too big Powerline symbols I also tried the `-l` flag with even worse result. I also checked if the problem was my Monaco font file. I downloaded @taohex's [one](https://github.com/taohex/fonts/blob/master/Monaco.dfont) but I got the same result. I guess it's the same as mine since the size is exactly the same. Anyway, you can download mine here [Monaco.dfont.zip](https://github.com/ryanoasis/nerd-fonts/files/1421007/Monaco.dfont.zip)

**Comment by ryanoasis** [comment]
@luisspuerto This is just a shot in the dark but what if you try the `--adjust-line-height` flag?

**Comment by luisspuerto** [comment]
I think I've already tried... but I can try again.

**Comment by luisspuerto** [comment]
the result is even worse. 

First screenshot if with the original Monaco and second is with @taohex's. The result is the same. 
![screen shot 2017-10-28 at 12 36 03](https://user-images.githubusercontent.com/22882880/32133277-6d221754-bbdd-11e7-820c-55f994b8dd1b.png)
![screen shot 2017-10-28 at 12 40 40](https://user-images.githubusercontent.com/22882880/32133278-6d42f794-bbdd-11e7-84ef-441c4280e7bf.png)

But the size of the final fonts aren't the same... as you can see here.
[Monaco patched.zip](https://github.com/ryanoasis/nerd-fonts/files/1424105/Monaco.patched.zip)

PS/ I've updated the gist, just in case you wanted to take a look to the output.

**Comment by ryanoasis** [comment]
@luisspuerto Ah okay darn. Thanks for the details and screenshots! :smile: 

Hopefully we can figure out what's happening :worried:

**Comment by luisspuerto** [comment]
Thanks! I hope so, because I've would like to have Monaco font patched. 
Have you tried to patch yourself? 

I've also emailed @taohex, but they didn't reply me yet.

**Comment by luisspuerto** [comment]
@ryanoasis by the way... can you try to patch yourself and see if the result is the same?  

Thanks!

**Comment by luisspuerto** [comment]
Hey! 

Any progress? I just did the patch in a computer with a clean install or high sierra and I have the same problem. It's like the height of lines increased for some reason... Perhaps some of the glyphs aren't adapting to the height. Probably the powerline ones.

**Comment by ryanoasis** [comment]
@luisspuerto Hey no I haven't tried patching myself. Also sorry no progress on this issue. Mostly getting caught up with the backlog of PRs (now finally merged). Next up will be the [v1.2.0 Release](https://github.com/ryanoasis/nerd-fonts/milestone/17)

**Comment by luisspuerto** [comment]
don't worry... and let's see if with the new release the issue fixes itself. 😊

**Comment by taigacute** [comment]
@ryanoasis  I don't think it solved. Is it reopened?Monaco has some problem, I got such a graph, the size of many graphics is not normal, specifically it is bigger。
![555](https://user-images.githubusercontent.com/41671631/52159685-b99c3f80-26e2-11e9-9e17-8ee1370f6548.png)
i test `e0b0` `e0b1` `e0b2` `e0b3` `e0b9` `e0bf` `e0c0` `e0c2` `e0c1` `e0c3` `e0bb` `e0bd`
`e0cc` `e0c8` `e0b6`  `e0b8` `e0ba` `e0c6` `e0b4` `e0bc` `e0be`   ,
I found these characters, their performance is not normal, on the statusline, these graphics will exceed a part。
![66](https://user-images.githubusercontent.com/41671631/52159745-a342b380-26e3-11e9-8a7b-9c30785106b4.png)
![88](https://user-images.githubusercontent.com/41671631/52159746-a3db4a00-26e3-11e9-8be0-72762346c909.png)

**Comment by luispuerto** [comment]
as far as I know it's not working... I usually tweak the result with font-forge app, but it's not perfect. 

It's pretty clear that something it's going on and the result monaco+nerds fonts doesn't have the same height as Monaco.

**Comment by taigacute** [comment]
> as far as I know it's not working... I usually tweak the result with font-forge app, but it's not perfect.
> 
> It's pretty clear that something it's going on and the result monaco+nerds fonts doesn't have the same height as Monaco.

@luispuerto  yes,Unable to patch it, this issue should remain open until it is fixed

**Comment by taigacute** [comment]
@ryanoasis  
I have noticed this problem for a long time, is it necessary to fix the monaco font?

**Comment by JoveYu** [comment]
any update? really need monaco patched font

**Comment by luispuerto** [comment]
I don't know if in the last version of the script whether this is fixed. I got mine patched and them I fixed it a little bit, so I was able to use it, but nothing more.

**Comment by JoveYu** [comment]
I use latest version to patch font, but not work ,the powerline symbols still too big, @luispuerto can you upload you patched monaco font?

**Comment by luispuerto** [comment]
@JoveYu I can't upload it, since Monaco has a private license. 

You can contact me and I can share with you, thought.

**Comment by JoveYu** [comment]
@luispuerto OK Thank you,  please check you Inbox of you email

**Comment by fragilelambda** [comment]
@JoveYu @luispuerto I met the same issue here, would you guys share that with me please? Thank you in advance!

**Comment by luispuerto** [comment]
@fragilelambda I no longer use that font since there was another issues. I'm using meslo that is really similar.

**Comment by luispuerto** [comment]
@ryanoasis I really don't know if I should close the issue. I no longer have it, since I'm no longer use the monaco patched font so I really don't know the issue it's still there or not. 

I also find it really pointless to have an issue here that if it's still happening you aren't going to fix for whatever reason. If this is the case, just label it as `won't fix` and close it.

**Comment by ryanoasis** [comment]
Hey I know it's been a very long time. I just haven't had the time to catch up. I know we need a better system of at least triaging the issues so there is at least some organizing. I feel like something like this should still attempt to be fixed.

**Comment by qiuwei** [comment]
I can confirm that this issue still persists with Monetery.

**Comment by Finii** [comment]
Ok, lets see which target fonts we have here

* Monaco
* Menlo
* Monetery

And I see two issues mentioned:

* Too big / high powerline synbols
* icons too small / far down

This should be fixable ;-) 

![image](https://user-images.githubusercontent.com/16012374/165110044-5d1647ba-7b58-4bcc-b564-e139bc3e76ac.png)

**Comment by Finii** [comment]
Hmm, with Menlo I can not reproduce (with current `font-patcher`).

The font does look ok:

![image](https://user-images.githubusercontent.com/16012374/165114182-a08f68c3-fdea-476c-b7e4-ee3e0ac44ce7.png)

This has been my Menlo source:

![image](https://user-images.githubusercontent.com/16012374/165114436-b2254bcd-8e49-4fcb-a907-0efaf1d7f0b2.png)

This is how I patched...:

![image](https://user-images.githubusercontent.com/16012374/165114714-cd9cf892-5b35-46f3-bc27-62aa2a84ea32.png)

Probably this has been fixed by #593.

Lets see if I can find the other fonts.
